apps/web/scripts/kernel/handlers/kernel.ts

```diff
@@ -1,26 +1,38 @@
 import { type MessengerCallback, RequestType } from '@walless/messaging';
 
+import * as coordinator from '../utils/coordinator';
+import { getNetwork } from '../utils/handler';
+
 import { handleConnect } from './connect';
 import * as solanaHandler from './solanaHandler';
 import * as suiHandler from './suiHandler';
 
 export const onKernelMessage: MessengerCallback = async (payload, channel) => {
-	if (payload.requestId) {
-		if (payload.type === RequestType.REQUEST_CONNECT) {
-			handleConnect(payload, channel);
-		} else if (payload.type == RequestType.GET_ENDPOINT_ON_SOLANA) {
+	const { type, requestId } = payload;
+
+	if (requestId) {
+		let handleMethod;
+		let requirePrivateKey = true;
```

Comment:
is there any way to define `requirePrivateKey` inside handleMethod, benefit: shorten this file, and inside the handle we have information of `requirePrivateKey` without have to open this file.